aten/src/ATen/native/cpu/SoftMaxKernel.cpp

```diff
@@ -219,9 +219,15 @@ inline void _vec_softmax(
   int64_t outer_stride = dim_size * dim_stride;
   int64_t grain_size = std::min(internal::GRAIN_SIZE / dim_size, (int64_t)1);
   int vectorized_step = Vec().size(); // Currently, we only support scalar_t with double or float32
+#ifdef CPU_CAPABILITY_AVX512
+  TORCH_CHECK(
+    (vectorized_step == 16) || (vectorized_step == 8),
```

Comment:
This doesn't look controllable by user input; `TORCH_INTERNAL_ASSERT`?

Reply:
Thank you! This `TORCH_CHECK` is already present in the master branch, but I'll change it to `TORCH_INTERNAL_ASSERT` at 2 places (I added one occurrence).